test/integration/certexpiry/certexpiry_test.go

```diff
@@ -0,0 +1,165 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package defserver
+
+import (
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/cloudflare/cfssl/config"
+	"github.com/cloudflare/cfssl/log"
+	"github.com/hyperledger/fabric-ca/cmd/fabric-ca-client/command"
+	"github.com/hyperledger/fabric-ca/internal/pkg/util"
+	"github.com/hyperledger/fabric-ca/lib"
+	"github.com/hyperledger/fabric-ca/lib/metadata"
+)
+
+const (
+	cmdName = "fabric-ca-client"
+)
+
+var (
+	defaultServer          *lib.Server
+	defaultServerPort      = 7055
+	defaultServerEnrollURL = fmt.Sprintf("http://admin:adminpw@localhost:%d", defaultServerPort)
+	defaultServerHomeDir   = "certExpiryServerDir"
+	storeCertsDir          = "/tmp/testCertsCertExpiry"
+	clientCAHome           = "/tmp/certExpiryCaHome"
+)
+
+func TestMain(m *testing.M) {
+	var err error
+
+	metadata.Version = "1.1.0"
+	os.Setenv("FABRIC_CA_SERVER_SIGNING_DEFAULT_EXPIRY", "1m")
+	os.Setenv("FABRIC_CA_CLIENT_HOME", clientCAHome)
+
+	os.RemoveAll(defaultServerHomeDir)
+	os.RemoveAll(storeCertsDir)
+	os.RemoveAll(clientCAHome)
+	defaultServer, err = getDefaultServer()
+	if err != nil {
+		log.Errorf("Failed to get instance of server: %s", err)
+		os.Exit(1)
+	}
+
+	err = defaultServer.Start()
+	if err != nil {
+		log.Errorf("Failed to start server: %s", err)
+		os.Exit(1)
+	}
+
+	rc := m.Run()
+
+	err = defaultServer.Stop()
+	if err != nil {
+		log.Errorf("Failed to stop server: %s, integration test results: %d", err, rc)
+		os.Exit(1)
+	}
+
+	os.RemoveAll(defaultServerHomeDir)
+	os.RemoveAll(storeCertsDir)
+	os.RemoveAll(clientCAHome)
+	os.Exit(rc)
+}
+
+func TestReenrollExpiredCert(t *testing.T) {
+	var err error
+
+	// Enroll a user that will be used for subsequent certificate commands
+	err = command.RunMain([]string{cmdName, "enroll", "-u", defaultServerEnrollURL, "-d"})
+	util.FatalError(t, err, "Failed to enroll user")
+
+	// Register a new user
+	err = command.RunMain([]string{cmdName, "register", "-u", defaultServerEnrollURL, "-d", "--csr.keyrequest.reusekey", "--id.name", "user1", "--id.secret", "user1pw", "--id.type", "client"})
+	util.FatalError(t, err, "Failed to register new user1")
+
+	userServiceEnrollURL := fmt.Sprintf("http://user1:user1pw@localhost:%d", defaultServerPort)
+
+	// Enroll and then reenroll to check
+	err = command.RunMain([]string{cmdName, "enroll", "-u", userServiceEnrollURL, "-d"})
+	util.FatalError(t, err, "Failed to enroll user1")
+
+	err = command.RunMain([]string{cmdName, "reenroll", "-u", userServiceEnrollURL, "-d", "--csr.keyrequest.reusekey"})
+	util.FatalError(t, err, "Failed to reenroll user1")
+
+	log.Infof("Tested re-enroll of id, waiting for cert to expiry before testing re-enroll\n")
+	time.Sleep(2 * time.Minute)
```

Comment:
could we make the expiry in seconds rather than a minute? I hate to sleep for two whole minutes in a test... they add up...

Reply:
agree with you; didn't find sub minute timeouts reliable in testing however..